AddsHTTPS support for REST protocol for AWS Neptune.

bhsu22 opened the pull request:
In trying to connect to a AWS Neptune remote database, I had to change the REST server url to `https` It seems like the new version of Neptune uses https now by default. This PR adds this to `graphioGremlin.js` currently, but I could see placing this in `graphConf.js` if that's the preferred method.

Some information if others are encountering the same issue (or maybe I did something horribly wrong). Users can tell if their Neptune cluster responds to https or http requests by trying the following commands (in their Bastion ec2 instance).
- `curl http://<your endpoint>:8182/status`
- `curl https://<your endpoint>:8182/status` 

After following instructions in [1], users could also try (in their browser on their local machine, __after__ setting up the ssh tunnel through their bastion host) the links `http://localhost:8182/status` or `https://localhost:8182/status` to diagnose this issue. For me, only https worked.

1. [Link](https://aws.amazon.com/premiumsupport/knowledge-center/neptune-gremlin-graph-graphexp/) to connecting a local graphexp instance to remote AWS Neptune server via a Bastion Host.

bricaud commented:
Thanks for this PR @bhsu22 . Yes it would be better to have it in the conf file graphConf.js as an option to use http or https. Could you make another PR in that sense? I do not use Neptune and I can't test it myself. There is another place where http might be replaced by https: in graphexp.html for loading d3 and jquery. That would be great if you can test the best way to do it with Neptune and make a PR.

bhsu22 commented:
Apologies for the delay, but I've added a flag to `graphConf.js` so that users can switch on using `https` vs `http` depending on the service that they're querying. The default is `REST_USE_HTTPS=false` to preserve the original behavior. Also I added a note about switching it to `true` for use with Neptune.

I didn't really see the place to make a similar change in `graphexp.html`. Did you have something in mind where https could be used instead of https? This is mostly to handle queries to the backend graph database and by my reading, it seems like that happens in `graphioGremlin.js`?

bricaud commented:
Thanks!
for the https in graphexp.html I was thinking where d3.js and jquery are loaded (they are loaded from an http address). But nevermind, it is fine like this.
